company_website/tests/test_send_email_utils.py

```diff
@@ -0,0 +1,95 @@
+        returned_recipients_list = _get_estimate_project_email_recipients()
+
+        self.assertEqual(returned_recipients_list, settings.DEFAULT_EMAIL_RECIPIENTS)
+
+    def test_that_when_recipient_list_is_not_empty_and_default_recipient_settings_is_disabled_it_return_recipients_list(
+        self,
+    ):
+        defined_recipient = "some_recipient@codepoets.it"
+
+        EstimateProjectEmailRecipients(name=defined_recipient).save()
+
+        returned_recipients_list = list(_get_estimate_project_email_recipients())
+
+        self.assertEqual(returned_recipients_list, [(defined_recipient,)])
+
+
+class SendEmailTestCase(TestCase):
+    def setUp(self) -> None:
+        self.template = ESTIMATE_PROJECT_EMAIL_TEMPLATE_NAME
+        self.template_context = {
+            "name": "John Doe",
+            "email": "email@address.com",
+            "idea_description": "Some description",
+            "privacy_policy_accepted": True,
+        }
+
+        self.expected_mail_body = (
+            "Project Liberation - New estimate project form has been filled in.\n\n"
+            "Form data:\n\n"
+            "Name: John Doe\n\n"
+            "Email address: email@address.com\n\n"
+            "Country:\n\n"
+            "Idea description: Some description\n\n"
+            "Is NDA required:\n\n"
+            "You want to\n\n"
+            "What is your monthly budget?\n\n"
+            "How much time do you have?\n\n"
+            "Do you want us to design your product?\n\n"
+            "How did you find us?:"
+        )
+
+    def test_that_create_messages_to_mail_method_create_proper_email_messages(self):
+        plain_text_message = _create_messages_to_mail(self.template, self.template_context)[1]
+
+        self.assertEqual(plain_text_message, self.expected_mail_body)
+
+    @parameterized.expand(
+        [
+            ("<p> Some text </p>", "Some text"),
+            ("<p> Some <a href='https://www.google.com/earth/'>text</p>", "Some text"),
+            ("<p> Some <em>text</em> </p>", "Some text"),
+        ]
+    )
+    def test_that_convert_html_to_plain_test_method_create_proper_message(
+        self, html_message, expected_plain_text_message
+    ):
+        returned_value = _convert_html_to_plain_text(html_message)
+
+        self.assertEqual(returned_value, expected_plain_text_message)
+
+    def test_that_send_email_to_management_pass_proper_parameters_to_send_email_method(self):
+        send_mail_to_management(self.template_context, ESTIMATE_PROJECT_EMAIL_SUBJECT, self.template)
+
+        mail_subject = mail.outbox[0].subject
+        mail_recipient = mail.outbox[0].to[0]
+        mail_body = mail.outbox[0].body
+
+        self.assertEqual(mail_subject, ESTIMATE_PROJECT_EMAIL_SUBJECT)
+
+        self.assertEqual(mail_recipient, settings.DEFAULT_EMAIL_RECIPIENTS[0])
+
+        self.assertEqual(mail_body, self.expected_mail_body)
```

Comment:
This is not how this test should be written. You should mock send_mail function and see if it was correctly called instead of changing mailbox. Right now we using as a default console mailing but if I would run this on my local env with production settings it might try to send emails.

Reply:
Done